Client/Helpers/FxALoginHelper.swift

```diff
@@ -184,13 +217,15 @@ class FxALoginHelper {
 
     func application(_ application: UIApplication, canDisplayUserNotifications allowed: Bool) {
         guard allowed else {
+            apnsTokenDeferred?.fill(Maybe.failure(PushNotificationError.userDisallowed))
             return readyForSyncing()
         }
 
         // Record that we have asked the user, and they have given an answer.
         profile?.prefs.setBool(true, forKey: applicationDidRequestUserNotificationPermissionPrefKey)
 
         guard #available(iOS 10, *) else {
+            apnsTokenDeferred?.fill(Maybe.failure(PushNotificationError.wrongOSVersion))
```

Comment:
For users who are on 9 shouldnt we just not show the extra slide? That way we dont have to show them an error?
This is probably irrelevant if this feature is landing in 9.0 though because iOS9 wouldnt be supported.